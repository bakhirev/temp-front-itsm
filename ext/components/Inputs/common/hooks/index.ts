export * from './useSuffixHook';
export * from './useTooltipHook';
