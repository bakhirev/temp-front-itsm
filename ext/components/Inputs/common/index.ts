export * from './interface';
export * from './hooks';
export * from './utils';
export * from './helpers';
export * from './types';
