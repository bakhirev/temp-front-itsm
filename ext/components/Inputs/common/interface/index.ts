export * from './iInputDefaultProps';
export * from './iStyledIconDefaultProps';
