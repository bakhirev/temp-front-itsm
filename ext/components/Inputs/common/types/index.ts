/** Типы для компонентов инпут */

export type Size = 'big' | 'small' | 'micro';
export type Status = 'success' | 'error' | 'default';
export type Mask = 'card' | 'phone';
export type Type = 'text' | 'password' | 'suggest' | 'number' | 'range' | 'informer';
