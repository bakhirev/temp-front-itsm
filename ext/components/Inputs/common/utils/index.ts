export * from './formatValue';
export * from './utils-style';
export * from './getIconSize';
