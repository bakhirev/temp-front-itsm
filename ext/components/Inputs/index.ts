import { TimePicker } from '../TimePicker';

import { InputText } from './Text';
import { InputPassword } from './Password';
import { InputSuggest } from './Suggest';
import { InputInformer } from './Informer';
import { InputNumber } from './Number';
import { InputMask } from './Mask';
import { InputRange } from './InputRange';
import { InputPhoneCountry } from './PhoneCountry';
export { IInputProps } from './BaseField';
export { IInputMaskProps } from './Mask';
export type { IOptionSuggest } from './Suggest';

export const Input = {
  Text: InputText,
  Password: InputPassword,
  Number: InputNumber,
  Mask: InputMask,
  Suggest: InputSuggest,
  Range: InputRange,
  Informer: InputInformer,
  PhoneCountry: InputPhoneCountry,
  Time: TimePicker,
};
