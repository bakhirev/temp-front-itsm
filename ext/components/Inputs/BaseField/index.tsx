import React, { useState, Dispatch, SetStateAction, ReactNode, FC } from 'react';

import { createTooltipHOC } from '../../TooltipHOC';
import { Body2Short } from '../../Typography';
import { IInputDefaultProps, useTooltipHook } from '../common';
import type { Type } from '../common/types';
import { DEFAULT_WIDTH } from '../constants';

import { InputSkeleton } from './InputSkeleton';
import { Label, Wrapper, Separator, StyledAdditionalText } from './StyledComponents';

export interface IInputProps extends IInputDefaultProps {
  /** Коллбэк на клик */
  onClick?: (event: any) => void | undefined;
  /** иконки из Figma , которые прокидываются в инпут */
  iconServices?: ReactNode;
  /** Отображение подсказки при переполнении инпута. По умолчанию false */
  withTooltip?: boolean;
  /** Тип инпута */
  type?: Type;
  /** Активна ли иконка в инпуте */
  iconIsActive?: boolean;
  /** Функция для переключения active icon */
  setIconIsActive?: Dispatch<SetStateAction<boolean>>;
  /** Использование числового инпута */
  number?: boolean;
  /** Использование Input Range */
  range?: boolean;
  /** Активен ли информер в инпуте */
  activeInformer?: boolean;
  /** Значение инпута */
  value?: any;
  /** Флаг ввода дробной части */
  withCoins?: boolean;
  phoneCountry?: boolean;
  countryIcon?: any;
  selectIcon?: any;
  onCountrySelectOpen?: Function;
  countrySelectComponent?: any;
  onCountrySelect?: Function;
  /** Дефолтное value */
  defaultValue?: any;
}

export const Input: FC<IInputProps> = ({
  disabled,
  label,
  additionalText,
  status = 'default',
  className,
  dataTestId,
  withTooltip = false,
  value = '',
  width = DEFAULT_WIDTH,
  refInput,
  tooltipContainer,
  ...props
}) => {
  const [focused, setFocused] = useState(false);
  const { tooltip, ref } = useTooltipHook(refInput, withTooltip, value);

  const Input = tooltip && !focused ? createTooltipHOC(InputSkeleton) : InputSkeleton;
  const inputElement = (
    <Input
      tooltip={value}
      container={tooltipContainer}
      display="block"
      disabled={disabled}
      focused={focused}
      setFocused={setFocused}
      status={status}
      value={value}
      ref={ref}
      width={width}
      {...props}
    />
  );

  return (
    <Wrapper
      disabled={disabled}
      focused={focused}
      className={className}
      data-test-id={dataTestId}
      width={width}
    >
      <Label>
        {label && (
          <>
            <Body2Short>{label}</Body2Short>
            <Separator />
          </>
        )}
        {inputElement}
      </Label>
      {additionalText && (
        <StyledAdditionalText disabled={disabled} status={status} focused={focused}>
          <Body2Short>{additionalText}</Body2Short>
        </StyledAdditionalText>
      )}
    </Wrapper>
  );
};
