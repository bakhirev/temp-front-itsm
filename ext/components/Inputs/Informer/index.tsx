import React, { useState, ReactNode, FC } from 'react';

import { ReactComponent as HelpSolid } from '../../Icons/service/HelpSolid.svg';
import { createHintHOC } from '../../HintHOC';
import { Input } from '../BaseField';
import { getIconSize } from '../common';
import type { IInputDefaultProps } from '../common/interface';

import { StyledQuestionSolidIcon } from './StyledQuestionSolidIcon';

export interface IInputInformer extends IInputDefaultProps {
  /** Значение инпута */
  value?: string;
  /** Текст информера */
  message?: string;
  /** Дополнительная иконка */
  icon?: ReactNode;
}

export const InputInformer: FC<IInputInformer> = ({
  value = '',
  disabled,
  size = 'big',
  message = '',
  ...props
}) => {
  const [visible, setVisible] = useState(false);

  const iconSize = getIconSize(size);
  const IconWrapper = ({ onClick }: { onClick: () => void }): JSX.Element => (
    <StyledQuestionSolidIcon active={visible} disabled={disabled} onMouseDown={onClick}>
      <HelpSolid width={iconSize} height={iconSize} />
    </StyledQuestionSolidIcon>
  );

  const ButtonWithHint = createHintHOC(IconWrapper);
  const handleRequestHide = () => {
    !disabled && setVisible(false);
  };

  const handleClick = () => {
    !disabled && setVisible(true);
  };

  const iconServices = (
    <ButtonWithHint
      message={message}
      offset={16}
      size={'small'}
      visible={!disabled ? visible : false}
      onClick={handleClick}
      onRequestHide={handleRequestHide}
    />
  );

  return (
    <Input
      size={size}
      disabled={disabled}
      iconServices={iconServices}
      activeInformer={visible}
      value={value}
      {...props}
    />
  );
};
