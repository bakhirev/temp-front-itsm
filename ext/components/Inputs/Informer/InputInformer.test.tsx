import React, { ChangeEvent } from 'react';
import { shallow } from 'enzyme';

import { Input } from '../../Inputs';
import { runCommonInputTest, runCommonInputOnChangeTest } from '../common/test';

jest.mock('../../common', () => ({
  DEFAULT_THEME: {},
  Z_INDEXES: {
    INFORMER: 777,
  },
}));

const textError = `Ошибка`;
const label = 'Информер';
const message = 'Привет информер';

describe('Input.Informer', () => {
  runCommonInputTest(Input.Informer);
  runCommonInputOnChangeTest(Input.Informer, '1000', '1000');

  it('should render component with message props', () => {
    const wrapper = shallow(
      <Input.Informer
        value={''}
        onChange={(event: ChangeEvent<HTMLInputElement>, value: string) => value}
        label={label}
        additionalText={textError}
        status={'error'}
        size={'big'}
        message={message}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
