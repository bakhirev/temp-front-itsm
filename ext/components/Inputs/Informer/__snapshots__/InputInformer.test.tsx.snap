// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input.Informer should render component 1`] = `
<Component
  activeInformer={false}
  iconServices={
    <HintHOC
      message=""
      offset={16}
      onClick={[Function]}
      onRequestHide={[Function]}
      size="small"
      visible={false}
    />
  }
  onChange={[Function]}
  size="big"
  value=""
/>
`;

exports[`Input.Informer should render component with clearable field, icon and name 1`] = `
<Component
  activeInformer={false}
  clearable={true}
  icon={<div />}
  iconServices={
    <HintHOC
      message=""
      offset={16}
      onClick={[Function]}
      onRequestHide={[Function]}
      size="small"
      visible={false}
    />
  }
  inputName="test"
  label="Введите ваше имя"
  onChange={[Function]}
  size="micro"
  value=""
/>
`;

exports[`Input.Informer should render component with error status 1`] = `
<Component
  activeInformer={false}
  additionalText="Ошибка"
  iconServices={
    <HintHOC
      message=""
      offset={16}
      onClick={[Function]}
      onRequestHide={[Function]}
      size="small"
      visible={false}
    />
  }
  label=""
  onChange={[Function]}
  size="big"
  status="error"
  value=""
/>
`;

exports[`Input.Informer should render component with message props 1`] = `
<Component
  activeInformer={false}
  additionalText="Ошибка"
  iconServices={
    <HintHOC
      message="Привет информер"
      offset={16}
      onClick={[Function]}
      onRequestHide={[Function]}
      size="small"
      visible={false}
    />
  }
  label="Информер"
  onChange={[Function]}
  size="big"
  status="error"
  value=""
/>
`;

exports[`Input.Informer should render component with success status 1`] = `
<Component
  activeInformer={false}
  additionalText="Успех"
  iconServices={
    <HintHOC
      message=""
      offset={16}
      onClick={[Function]}
      onRequestHide={[Function]}
      size="small"
      visible={false}
    />
  }
  label="Введите ваше имя"
  onChange={[Function]}
  size="big"
  status="success"
  value=""
/>
`;
